contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,807 @@
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getChildIterator(String plane, PointerByReference iter) {
+            return INSTANCE.IORegistryEntryGetChildIterator(this, plane, iter);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetChildEntry} to return the
+         * first child of this registry entry in a plane.
+         *
+         * @param plane
+         *            The name of an existing registry plane.
+         * @param child
+         *            The first child of the registry entry, on success. The child must
+         *            be released by the caller.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getChildEntry(String plane, PointerByReference child) {
+            return INSTANCE.IORegistryEntryGetChildEntry(this, plane, child);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryGetParentEntry} to return the
+         * first parent of this registry entry in a plane.
+         *
+         * @param plane
+         *            The name of an existing registry plane.
+         * @param parent
+         *            The first parent of the registry entry, on success. The parent
+         *            must be released by the caller.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int getParentEntry(String plane, PointerByReference parent) {
+            return INSTANCE.IORegistryEntryGetParentEntry(this, plane, parent);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryCreateCFProperty} to create a
+         * CF representation of this registry entry's property.
+         *
+         * @param key
+         *            A {@code CFString} specifying the property name.
+         * @return A CF container is created and returned the caller on success.
+         *         <p>
+         *         The caller should release with {@link CoreFoundation#CFRelease}.
+         */
+        public CFTypeRef createCFProperty(CFStringRef key) {
+            return INSTANCE.IORegistryEntryCreateCFProperty(this, key, CoreFoundation.INSTANCE.CFAllocatorGetDefault(),
+                    0);
+        }
+
+        /**
+         * Convenience method for {@link #IORegistryEntryCreateCFProperties} to create a
+         * CF dictionary representation of this registry entry's property table.
+         *
+         * @param properties
+         *            A CFDictionary is created and returned the caller on success. The
+         *            caller should release with CFRelease.
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int createCFProperties(PointerByReference properties) {
+            return INSTANCE.IORegistryEntryCreateCFProperties(this, properties,
```

Comment:
Same comment as in `getRegistryEntryID`. In this case an `CFMutableDictionaryRef` could be returned directly.